kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/AsyncProcessor.java

```diff
@@ -254,7 +254,7 @@ private void completeInitialization() {
         streamThreadName,
         taskId.partition(),
         connectedStoreBuilders.values(),
-        this::flushPendingEventsForCommit
+        asyncThreadPoolRegistration::flushAllAsyncEvents
```

Comment:
This is actually the only change that was strictly necessary for supporting async, and it addresses the issue raised in this earlier comment by simply flushing all the async processors (stateless and stateful) in topological order from inside #flushCache, rather than flushing the stateful async processors in their individual #flushCache calls and then later flushing all processors during the producer#flush